src/main/webapp/app/exercise/result/result.utils.spec.ts

```diff
@@ -47,114 +47,131 @@ describe('ResultUtils', () => {
+        { result: { score: MIN_SCORE_GREEN, testCaseCount: 1 }, participation: {} as Participation, templateStatus: ResultTemplateStatus.HAS_RESULT, expected: 'text-success' },
+        { result: { score: MIN_SCORE_ORANGE, testCaseCount: 1 }, participation: {} as Participation, templateStatus: ResultTemplateStatus.HAS_RESULT, expected: 'result-orange' },
+        { result: {}, participation: {} as Participation, templateStatus: ResultTemplateStatus.HAS_RESULT, expected: 'text-danger' },
         {
-            result: { score: 1, submission: { participation: { exercise: { type: ExerciseType.PROGRAMMING } } } } as Result,
+            result: { score: 1 } as Result,
+            participation: { exercise: { type: ExerciseType.PROGRAMMING } as Exercise, submissions: [{ id: 1 }] } as Participation,
             templateStatus: ResultTemplateStatus.HAS_RESULT,
             expected: 'text-success',
         },
-    ])('should correctly determine text color class', ({ result, templateStatus, expected }) => {
-        expect(getTextColorClass(result, templateStatus!)).toBe(expected);
+    ])('should correctly determine text color class', ({ result, participation, templateStatus, expected }) => {
+        expect(getTextColorClass(result, participation, templateStatus!)).toBe(expected);
     });
 
     it.each([
         {
-            result: { submission: { participation: { exercise: { type: ExerciseType.PROGRAMMING } } } } as Result,
+            result: {} as Result,
+            participation: { exercise: { type: ExerciseType.PROGRAMMING } as Exercise } as Participation,
             templateStatus: ResultTemplateStatus.HAS_RESULT,
             expected: faCheckCircle,
         },
-        { result: undefined, templateStatus: ResultTemplateStatus.HAS_RESULT, expected: faQuestionCircle },
+        { result: undefined, participation: {} as Participation, templateStatus: ResultTemplateStatus.HAS_RESULT, expected: faQuestionCircle },
         {
             result: { submission: { submissionExerciseType: SubmissionExerciseType.PROGRAMMING, buildFailed: true }, assessmentType: AssessmentType.AUTOMATIC },
+            participation: {} as Participation,
             templateStatus: ResultTemplateStatus.HAS_RESULT,
             expected: faTimesCircle,
         },
         {
-            result: { submission: { participation: { type: ParticipationType.PROGRAMMING, exercise: { type: ExerciseType.PROGRAMMING } } } } as Result,
+            result: {} as Result,
+            participation: { type: ParticipationType.PROGRAMMING, exercise: { type: ExerciseType.PROGRAMMING } as Exercise } as Participation,
             templateStatus: ResultTemplateStatus.HAS_RESULT,
             expected: faQuestionCircle,
         },
         {
             result: { score: undefined, successful: true, feedbacks: [{ type: FeedbackType.AUTOMATIC, text: 'This is a test case' }], testCaseCount: 1 },
+            participation: {} as Participation,
```

Comment:
It looks quite odd that we require participation to be defined and pass an empty object here for the test 🤔 
I doubt that it really needs to be defined